### PR TITLE
Prometheus - Docs and templates for private files

### PR DESCRIPTION
**Pull request by odscjames**
https://github.com/open-contracting/deploy/issues/31

### CODE REVIEW

**Comment by odscjames** [comment]
[Set up OCP Prometheus server](https://trello.com/c/OcivonLE/118-set-up-ocp-prometheus-server)

File: docs/reference/index.rst [hunk 1]
```diff
@@ -32,9 +32,22 @@ For access details, check the configuration file ``pillar/private/prometheus_pil
...
-Currently, Open Data Services runs a Prometheus server to process client data, which raises alarms to ODS staff only (`#31 <https://github.com/open-contracting/deploy/issues/31>`__).
+The alerting service handles actually sending alarms. Alarms are raised to relevant ODS staff and OCP staff.
+
+In the alerting service you can also put in a temporary "silence", with definitions of what alarms will be silenced. This can be used to avoid alerts if you know you are about to do something that would generate an alarm, such as taking a server off-line.
+
+To add a new server to Prometheus:
+
+-  Make sure ``prometheus-client-apache`` or ``prometheus-client-nginx`` is included for the server in ``salt/top.sls``.
+-  Salt will try to work out a domain name to use for the client server automatically. If it gets this wrong, you can set one manually by setting the ``prometheus.client_fqdn`` variable. Make sure you set this variable for one server only, and not all servers!
+-  Deploy to the server you want to moniter. This should now be working - test this by going to the end point in a web browser.
```
Comment (jpmckinney) [inline docs/reference/index.rst]: Typos: moniter -> monitor, end point -> endpoint (again below)

Comment (jpmckinney) [reply docs/reference/index.rst]: What is the endpoint? Can you add a link or a sample URL?

File: docs/reference/index.rst [hunk 2]
```diff
@@ -32,9 +32,22 @@ For access details, check the configuration file ``pillar/private/prometheus_pil
...
 
-Currently, Open Data Services runs a Prometheus server to process client data, which raises alarms to ODS staff only (`#31 <https://github.com/open-contracting/deploy/issues/31>`__).
+The alerting service handles actually sending alarms. Alarms are raised to relevant ODS staff and OCP staff.
+
+In the alerting service you can also put in a temporary "silence", with definitions of what alarms will be silenced. This can be used to avoid alerts if you know you are about to do something that would generate an alarm, such as taking a server off-line.
+
+To add a new server to Prometheus:
+
+-  Make sure ``prometheus-client-apache`` or ``prometheus-client-nginx`` is included for the server in ``salt/top.sls``.
+-  Salt will try to work out a domain name to use for the client server automatically. If it gets this wrong, you can set one manually by setting the ``prometheus.client_fqdn`` variable. Make sure you set this variable for one server only, and not all servers!
```
Comment (jpmckinney) [inline docs/reference/index.rst]: In what cases does it get it wrong? Does it currently get it wrong? How can we tell if it's wrong?

File: pillar/private-templates/prometheus_pillar.sls [hunk 3]
```diff
@@ -3,4 +3,15 @@ prometheus:
   # use "openssl passwd -apr1" to generate this. Use the same password as above.
   client_password_as_nginx_file: "prom:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
   client_fqdn:
+  client_port: 80
```
Comment (jpmckinney) [inline pillar/private-templates/prometheus_pillar.sls]: Aren't some of these not private? Same question about others.

**Comment by odscjames** [comment]
> Aren't some of these not private? Same question about others.

Yeah, technically a lot of what is there isn't secret like passwords. But in our own Prometheus installation, we raised a discussion about the fact that having people knowing the exact limits and things we are monitoring might be bad if people who are deliberately trying to mess with our servers can use that knowledge to avoid tripping our alarms. As the grumpy people say, "Obscurity is not security" is an incomplete saying, it's really "Obscurity is not security if that is the ONLY security measure you have". Sometimes alongside security a bit of obscurity can be good too. Anyway, we decided to just keep a lot of our Prom setup private until we could decide how to settle this. 

But this pull request just puts in place docs that reflect what we currently have - can we put this in now, and have a separate issue to discuss making some of the currently private setup public, if that's something you want to do?

I'll do the other points now.

**Comment by odscjames** [comment]
Updated

**Comment by odscjames** [comment]
@jpmckinney Are you happy with this update? Do merge if so.

**Pull request merged**